### PR TITLE
Fixes to the renderer list

### PR DESCRIPTION
**Pull request by martint-unity**
---
### Purpose of this PR
Fixes for this Bug
https://fogbugz.unity3d.com/f/cases/1281007/

made it reordereable,
Update the default renderer index properly when reordering
Able to delete entries and update indexes properly

renderer_list_fixes

Added a button to get to the Render Pipeline Asset if the index is missing from the list of renderers.
image

Also fixed an undo issue when removing a renderer from the list.

### CODE REVIEW

**Review by ellioman — APPROVED** [review]

**Review by Verasl — CHANGES_REQUESTED** [review]
Fix the default button overlap and you get my vote 👍 
image

**Review by Verasl — APPROVED** [review]
LGTM

**Review by erikabar — APPROVED** [review]
no issues found, tested OSX and Windows.

**Review by ellioman — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]
LGTM for fixing the issue. However we should discuss possibility of referencing renderers in GUIds in the camera instead of indices as there are multiple problems with this (what happens if you switch a pipeline asset that has different default renderers / indices?)

File: com.unity.render-pipelines.universal/Runtime/UniversalAdditionalCameraData.cs [hunk 1]
```diff
@@ -160,6 +160,18 @@ internal static UniversalAdditionalCameraData defaultAdditionalCameraData
             }
         }
 
+        #if UNITY_EDITOR
+        internal new Camera camera
```
[the remark this answers is not in the document]
Comment (martint-unity) [reply com.unity.render-pipelines.universal/Runtime/UniversalAdditionalCameraData.cs:164]: In the log
Debug.LogWarning(
                        $"Renderer at <b>index {m_RendererIndex.ToString()}</b> is missing for camera <b>**{camera.name}**</b>

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 2]
```diff
@@ -102,8 +102,9 @@ public partial class UniversalRenderPipelineAsset : RenderPipelineAsset, ISerial
         ScriptableRenderer[] m_Renderers = new ScriptableRenderer[1];
 
         // Default values set when a new UniversalRenderPipeline asset is created
-        [SerializeField] int k_AssetVersion = 5;
-        [SerializeField] int k_AssetPreviousVersion = 5;
+        const int k_MaxAssetVersion = 6;
```
[the remark this answers is not in the document]
Comment (martint-unity) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:105]: Yes true, that is my other PR. Can remove this.

File: com.unity.render-pipelines.universal/Runtime/UniversalAdditionalCameraData.cs [hunk 3]
```diff
@@ -160,6 +160,18 @@ internal static UniversalAdditionalCameraData defaultAdditionalCameraData
...
+        {
+            get => gameObject.GetComponent<Camera>();
```
[the remark this answers is not in the document]
Comment (martint-unity) [reply com.unity.render-pipelines.universal/Runtime/UniversalAdditionalCameraData.cs:166]: Done.

File: com.unity.render-pipelines.universal/Runtime/UniversalAdditionalCameraData.cs [hunk 4]
```diff
@@ -294,7 +306,18 @@ public bool requiresColorTexture
         /// </summary>
         public ScriptableRenderer scriptableRenderer
         {
-            get => UniversalRenderPipeline.asset?.GetRenderer(m_RendererIndex);
+            get
+            {
+                if (!UniversalRenderPipeline.asset.ValidateRendererData(m_RendererIndex))
```
[the remark this answers is not in the document]
Comment (martint-unity) [reply com.unity.render-pipelines.universal/Runtime/UniversalAdditionalCameraData.cs:311]: Done

**Review by Verasl — APPROVED** [review]
LGTM

**Review by erikabar — APPROVED** [review]
Tested with URP template (reordering renderers, removing, undo, redo, setting default) and UniversalRenderingExamples project

**Comment by ellioman** [comment]
Yamato issues on HDRP unrelated to this PR.

**Pull request merged**